crates/mun_codegen/src/intrinsics.rs

```diff
@@ -17,8 +18,8 @@ pub trait Intrinsic: Sync {
 }
 
 intrinsics! {
-    /// Allocates memory from the runtime to use in code.
-    pub fn malloc(size: u64, alignment: u64) -> *mut u8;
+    /// Allocates memory for the specified type.
+    pub fn new(type: *const TypeInfo, alloc_handle: *mut ffi::c_void) -> *const *mut ffi::c_void;
```

Comment:
We should explain what this alloc_handle actually is. Even though I wrote it, its unclear to me when I read it.